src/api.cc

```diff
@@ -547,8 +559,7 @@ CheckInResult AkliteClient::CheckInLocal(const LocalUpdateSource* local_update_s
     err_msg =
         "Couldn't find Targets matching the device's hardware ID; check a tag or a hardware ID of the device and the "
         "bundle's tag";
-    LOG_ERROR << err_msg;
-    return CheckInResult{CheckInResult::Status::NoMatchingTargets, "", {}};
+    return checkInFailure(client_, hw_id_, CheckInResult::Status::NoMatchingTargets, err_msg);
```

Comment:
The previous version didn't invoke `notifyTufUpdateFinished()` while `checkInFailure` does it unconditionally.

Reply:
It was missing before. We should call `notifyTufUpdateFinished()` at this point.

Reply:
Ah, ok.